Improve PHP security recommendations (#2423)
ibexa/documentation-developer: docs/infrastructure_and_maintenance/security/security_checklist.md

You should also make sure no stack trace is ever visible to end users of production sites,
though visible arguments are unsafe even if the stack traces only show up in log files.

### Disable error output from PHP

> Symfony in production mode will prevent exception messages from being visible to end users.

Author: Present tense:
```suggestion
Symfony in production mode prevents exception messages from being visible to end users.
```

However, if Symfony fails to boot properly, such exceptions may end up being visible, including stack traces.
This can be prevented by [disabling error message output in PHP](https://www.php.net/manual/en/language.errors.basics.php).
These `php.ini` configuration values should be used on production sites.

> When using Ibexa Cloud / Platform.sh the same settings can be configured in Ibexa DXP's `.platform.app.yaml` file.

Author: Fixing name error:
```suggestion
When using '[[= product_name_cloud =]]' / Platform.sh the same settings can be configured in '[[= product_name_base =]]' DXP's `.platform.app.yaml` file.
```

> When using '[[= product_name_cloud =]]' / Platform.sh the same settings can be configured in '[[= product_name_base =]]' DXP's `.platform.app.yaml` file.

Reviewer: ```suggestion
When using [[= product_name_cloud =]] the same settings can be configured in [[= product_name_base =]] DXP's `.platform.app.yaml` file.
```

Preview link: https://ez-systems-developer-documentation--2423.com.readthedocs.build/en/2423/infrastructure_and_maintenance/security/security_checklist/#symfony-production-mode
<img width="933" alt="Zrzut ekranu 2024-07-3 o 13 23 07" src="https://github.com/ibexa/documentation-developer/assets/10993858/9400186f-c8de-477e-9c4f-f01c8fcd7cc3">

No need to quote the variables, the end result does not look good.

I'm not sure if we should mention Platform.sh here - for us the product name is Ibexa Cloud and I'd focus on it

Reviewer 2: Continuing on @mnocon sugggestion, I would factorize the product name, and add a comma:
```suggestion
When using [[= product_name_cloud =]], the same settings can be configured in [[= product_name =]]'s `.platform.app.yaml` file.
```

    ```ini
    display_errors          = Off
    display_startup_errors  = Off

>     ```

Reviewer: Please remove the indent, it's not correctly formatted right now (https://ez-systems-developer-documentation--2423.com.readthedocs.build/en/2423/infrastructure_and_maintenance/security/security_checklist/#enable-zendexception_ignore_args-in-php-74-and-newer)

(I tried to create a suggestion from GitHub, but it doesn't handle nested backticks well 😅 )

Reviewer 2: ````suggestion
```ini
display_errors          = Off
display_startup_errors  = Off
```
````

### Other PHP settings

Consider what other security related settings are relevant for your needs.

> The [OWASP PHP Configuration Cheat Sheet](https://cheatsheetseries.owasp.org/cheatsheets/PHP_Configuration_Cheat_Sheet.html)

Reviewer: Seeing that it doesn't even acknowledge PHP 8 and was not updated in 4 years I'm torn if we should mention it to be honest

Author: Yes, same, but OWASP is such a great resource in general and this page links to others that are also good. I also expect the OWASP community to update this page eventually. I'm doubtfully leaning towards keeping it.

> contains some recommendations, but please beware it may be out of date as it doesn't mention PHP 8.

Reviewer 3: ```suggestion
contains several recommendations, but be aware that they may be out of date as they don't mention PHP 8.
```

Author: Sounds good!
